problem.schema.json

```diff
@@ -0,0 +1,122 @@
+                    "name": {
+                        "description": "The name of the variable. Every variable within the same stage must have a unique name.",
+                        "type": "string"
+                    },
+                    "stage": {
+                        "description": "The 0-indexed stage that the variable appears as a decision in.",
+                        "type": "integer",
+                        "minimum": 0
+                    },
+                    "obj": {
+                        "description": "An expression for the linear objective coefficient of the variable.",
+                        "$ref": "#/definitions/AffineFunction"
+                    },
+                    "lb": {
+                        "description": "An expression for the lower bound of the variable.",
+                        "$ref": "#/definitions/AffineFunction"
+                    },
+                     "ub": {
+                        "description": "An expression for the upper bound of the variable.",
+                        "$ref": "#/definitions/AffineFunction"
+                    },
+                    "type": {
+                        "description": "The type of the variable.",
+                        "enum": ["CONTINUOUS", "BINARY", "INTEGER"]
+                    }
+                 }
+            }
+        },
+        "constraints": {
+            "description": "An array containing the constraints of the problem.",	
+            "type": "array",
+            "items":  {
+                "type": "object",
+                  "required": ["type", "lhs", "rhs"],
+                 "properties": {
+                    "name": {
+                        "description": "The name of the constraint.",
+                        "type": "string"
+                    },
+                    "type": {
+                        "description": "The type of the inequality EQ for ==, LEQ for <=, and GEQ for >=.",
+                        "enum": ["EQ", "LEQ", "GEQ"]
+                    },
+                    "lhs": {
+                        "description": "An array containing the terms in the left-hand side of the constraints of the problem.",
+                        "type": "array",
+                        "items":  {
+                            "type": "object",
+                            "required": ["name", "stage", "coefficient"],
+                            "properties": {
+                                "name": {
+                                    "description": "The name of the variable.",
+                                    "type": "string"
+                                },
+                                "stage": {
+                                    "description": "The 0-based index of the stage in which the variable appears as a decision variable.",
+                                    "type": "integer",
+                                    "minimum": 0
+                                },
```

Comment:
Okay, so for this one: What dependency can be introduced? If the constraint is in stage `t`, can `x[t-2]` appear? If so, does the parser need to identify this and expand the state space?